pkg/platform/abstract/platform_test.go

```diff
@@ -946,6 +946,100 @@ func (suite *AbstractPlatformTestSuite) TestCreateFunctionEvent() {
 	suite.Require().NoError(err)
 }
 
+func (suite *AbstractPlatformTestSuite) TestValidateNodeSelector() {
+	for idx, testCase := range []struct {
+		nodeSelector         map[string]string
+		shouldFailValidation bool
+	}{
+		{
+			nodeSelector: map[string]string{
+				"some-key": "some-value",
+			},
+		},
+		{
+			nodeSelector: map[string]string{
+				"some-key": "",
+			},
+		},
+		{
+			nodeSelector: map[string]string{
+				"beta.kubernetes.io/arch":                        "amd64",
+				"beta.kubernetes.io/os":                          "linux",
+				"feature.node.kubernetes.io/cpu-cpuid.ADX":       "true",
+				"feature.node.kubernetes.io/kernel-version.full": "3.10.0-1127.13.1.el7.x86_64",
+			},
+		},
+		{
+			nodeSelector: map[string]string{
+				"some-key": "_some-value",
+			},
+			shouldFailValidation: true,
+		},
+		{
+
+			// should fail on empty key ""
+			nodeSelector: map[string]string{
+				"": "some-value",
+			},
+			shouldFailValidation: true,
+		},
+		{
+
+			// should fail on value "_some-value"
+			nodeSelector: map[string]string{
+				"some-key":      "some-value",
+				"some-key/name": "_some-value",
+			},
+			shouldFailValidation: true,
+		},
+		{
+
+			// should fail on key "/some-key" (empty DNS subdomain)
+			nodeSelector: map[string]string{
+				"some-key":  "some-value",
+				"/some-key": "some-value",
+			},
+			shouldFailValidation: true,
+		},
+	} {
+
+		suite.mockedPlatform.On("GetProjects", &platform.GetProjectsOptions{
```

Comment:
use 

```
suite.Run(testCase.name, func() { ...})
```
to wrap each test case. that will benefit us with the option to run each test case separately